_rules/device-motion-user-interface-7677a9.md

```diff
@@ -0,0 +1,259 @@
+---
+id: 7677a9
+name: Device motion based functionality can also be operated from the user interface
+rule_type: atomic
+description: |
+  This rule checks that functionality that can be operated by device motion can also be operated by user interface components.
+accessibility_requirements:
+  wcag21:2.5.4: # Motion Actuation (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Accessibility tree
+acknowledgments:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+htmlHintIgnore:
+  # https://www.npmjs.com/package/htmlhint
+  # (used with `npm test` to ensure validity of code snippets)
+  - 'attr-lowercase'
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with an associated [Window object][] that has an [event listener list][] with one or more [event listeners][event listener] for [device orientation events][device orientation] or [device motion events][device motion].
+
+## Expectation 1
+
+For each registered [device orientation event][device orientation] or [device motion event][device motion] in the test target there exists a set of one or more [instruments][instrument] supporting indistinguishable results of those from the event.
+
+## Expectation 2
+
+For at least one set of [instruments][instrument] from Expectation 1, all of the [instruments][instrument] in such set meet one of the following conditions:
+
+- the [instrument][] is in the same [web page][] of the test target; or
+- the [instrument][] can be found in a [clearly labeled location][].
+
+**Note:** If the set of instruments has more than one instrument, not every instrument of the set needs to be located in the same location.
+
+## Assumptions
```

Comment:
You're making the assumption here that every page with these event handlers attached uses them to operate some functionality. That isn't necessarily true. For example if you're playing a game that uses device motion, the event handlers are likely attached before the game is started. The user may have to click "play" before the actual "functionality" appears. The instrument doesn't need to be available from the start, as long as it's available as soon as you're actually playing the game.

I think this is something that needs to be dealt with in this rule, and not something that can just be assumed. I think the cause of this is the difference between our own definition of HTML web page, which is the page in a single state, and WCAG's definition of Web page, which is that page in any state it can possibly be in.

Reply:
Hmmm… The expectation is that there are instruments to support the same results as the event.
If the event has no result (e.g. before pressing "Play"), then we need instruments to support no result, which may arguably be OK with no instruments at all…

Maybe all that is needed is to have Expectation 1 be explicit that if the events do nothing, no instrument is actually needed.

Reply:
Similar to #880 I'll update the expectation to include the possibility of the events not having any result on the test target.